assets/js/frontend/course-video/youtube-extension.js

```diff
@@ -0,0 +1,81 @@
+( () => {
+	// Use of Page Visibility API. Unfortunately, it isn't standardized.
+	// https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API
+	let hidden, visibilityChange;
+	if ( typeof document.hidden !== 'undefined' ) {
+		// Opera 12.10 and Firefox 18 and later support
+		hidden = 'hidden';
+		visibilityChange = 'visibilitychange';
+	} else if ( typeof document.msHidden !== 'undefined' ) {
+		hidden = 'msHidden';
+		visibilityChange = 'msvisibilitychange';
+	} else if ( typeof document.webkitHidden !== 'undefined' ) {
+		hidden = 'webkitHidden';
+		visibilityChange = 'webkitvisibilitychange';
+	}
+
+	function handleVisibilityChange( player ) {
+		return function () {
+			if ( document[ hidden ] ) {
+				player.pauseVideo();
+			}
+		};
+	}
+
+	function onYouTubePlayerStateChange( event ) {
+		const playerStatus = event.data;
+
+		if (
+			window.sensei.courseVideoSettings.courseVideoRequired &&
+			playerStatus === YT.PlayerState.ENDED
+		) {
+			document.querySelector(
```

Comment:
One thing to notice here and in the other JS file is that we can have multiple buttons added to the page. And the `querySelector` would get only the first one.

Reply:
Fixed here: https://github.com/Automattic/sensei/pull/4546/commits/cb548745abf9015726b086fa9881c3eff80f39c3